### PR TITLE
refactor(core): Introduce overload for number-type node parameter (no-changelog)

### PR DESCRIPTION
**Pull request by ivov**

### CODE REVIEW

File: packages/core/src/NodeExecuteFunctions.ts [hunk 1]
```diff
@@ -2423,7 +2423,7 @@ export function getExecuteFunctions(
 				constructExecutionMetaData,
 			},
 		};
-	})(workflow, runExecutionData, connectionInputData, inputData, node);
+	})(workflow, runExecutionData, connectionInputData, inputData, node) as IExecuteFunctions;
```
Comment (ivov) [inline packages/core/src/NodeExecuteFunctions.ts:2426]: Rationale explained [here](https://linear.app/n8n/issue/N8N-3004#comment-9d34f12b).

File: packages/workflow/src/Interfaces.ts [hunk 2]
```diff
@@ -549,7 +549,34 @@ export interface IGetNodeParameterOptions {
...
 }
 
-export interface IExecuteFunctions {
+namespace ExecuteFunctions {
+	namespace NumberReturning {
+		export type NodeParameter = 'limit';
+	}
+
+	export type GetNodeParameterFn = {
+		// @TECH_DEBT: Refactor to remove this barely used overload
```
Comment (netroy) [inline packages/workflow/src/Interfaces.ts:558]: can you create a ticket for this?

**Comment by netroy** [comment]
looks great. but, prettier changes make it a bit difficult to read what all changed.

File: packages/workflow/src/Interfaces.ts (hunk 2 above)
Comment (ivov) [reply packages/workflow/src/Interfaces.ts:558]: Link: https://linear.app/n8n/issue/N8N-5632/refactor-to-remove-overload-with-generic-for-getnodeparameter

**Comment by ivov** [comment]
My bad! Will prevent next time. 

For reference, these had a Prettier pass:

- packages/nodes-base/nodes/Asana/Asana.node.ts 
- packages/nodes-base/nodes/Aws/CertificateManager/AwsCertificateManager.node.ts
- packages/nodes-base/nodes/Dropbox/Dropbox.node.ts
- packages/nodes-base/nodes/Google/Gmail/v1/GmailV1.node.ts
- packages/nodes-base/nodes/Mautic/Mautic.node.ts
- packages/nodes-base/nodes/NocoDB/NocoDB.node.ts
- packages/nodes-base/nodes/Notion/v1/NotionV1.node.ts
- packages/nodes-base/nodes/Notion/v2/NotionV2.node.ts
- packages/nodes-base/nodes/Pipedrive/Pipedrive.node.ts
- packages/nodes-base/nodes/S3/S3.node.ts
- packages/nodes-base/nodes/SendGrid/SendGrid.node.ts
- packages/nodes-base/nodes/Spotify/Spotify.node.ts
- packages/nodes-base/nodes/Venafi/Datacenter/VenafiTlsProtectDatacenter.node.ts
- packages/nodes-base/nodes/Venafi/ProtectCloud/VenafiTlsProtectCloud.node.ts

**Comment by netroy** [comment]
LGTM, but there are conflicts now.

File: packages/workflow/src/Interfaces.ts [hunk 3]
```diff
@@ -573,6 +577,12 @@ namespace ExecuteFunctions {
 			fallbackValue?: any,
 			options?: IGetNodeParameterOptions,
 		): boolean;
+		getNodeParameter(
+			parameterName: NumberReturning.NodeParameter,
+			itemIndex: number,
+			fallbackValue?: any,
```
Comment (netroy) [inline packages/workflow/src/Interfaces.ts:583]:
```suggestion
			fallbackValue?: number,
```

File: packages/workflow/src/Interfaces.ts [hunk 4]
```diff
@@ -573,6 +577,12 @@ namespace ExecuteFunctions {
 			fallbackValue?: any,
```
Comment (netroy) [inline packages/workflow/src/Interfaces.ts:577]: might as well fix this 
```suggestion
			fallbackValue?: boolean,
```

**Review by netroy — APPROVED** [review]

**Comment by janober** [comment]
Got released with `n8n@0.203.1`

**Comment by janober** [comment]
Got released with `n8n@0.204.0`

**Pull request merged**